src/main/java/com/avispl/symphony/dal/avdenvices/encoderdecoder/haivision/kraken/HaivisionKrakenCommunicator.java

```diff
@@ -0,0 +1,802 @@
+				return;
+			}
+			JsonNode responseMetadata = this.doGet(HaivisionCommand.GET_METADATA, JsonNode.class);
+			if (responseMetadata != null && responseMetadata.has(HaivisionConstant.METADATA_LIST) && responseMetadata.get(HaivisionConstant.METADATA_LIST).isArray()) {
+				// Loop through the metadata UUIDs from the original object
+				for (JsonNode metadataUuidNode : node) {
+					String metadataUuid = metadataUuidNode.asText();
+					for (JsonNode metadataItem : responseMetadata.get(HaivisionConstant.METADATA_LIST)) {
+						String metadataID = metadataItem.get(HaivisionConstant.UUID).asText();
+						if (metadataUuid.equals(metadataID)) {
+							String metadataName = metadataItem.get(HaivisionConstant.NAME).asText();
+							stats.put("Stream: " + name + HaivisionConstant.HASH + "Metadata", metadataName);
+						}
+					}
+				}
+			}
+		} catch (Exception e) {
+			logger.error("Error while populating the metadata Info", e);
+		}
+	}
+
+
+	/**
+	 * {@inheritDoc}
+   */
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> controllableProperties) throws Exception {
+		if (CollectionUtils.isEmpty(controllableProperties)) {
+			throw new IllegalArgumentException("ControllableProperties can not be null or empty");
+		}
+		for (ControllableProperty p : controllableProperties) {
+			try {
+				controlProperty(p);
+			} catch (Exception e) {
+				logger.error(String.format("Error when control property %s", p.getProperty()), e);
+			}
+		}
+	}
+
+	/**
+	 * Authenticates the user by sending a login request and retrieves the session cookie or ID.
+	 *
+	 * @return {@code true} if the session is successfully retrieved, {@code false} otherwise.
+	 * @throws ResourceNotReachableException if an error occurs during the login process.
+	 */
+	private boolean getCookieSession() {
+		try {
+			HttpClient client = this.obtainHttpClient(true);
+			HttpPost httpPost = new HttpPost(buildDeviceFullPath(HaivisionCommand.API_LOGIN));
+
+			String jsonPayload = String.format(HaivisionConstant.AUTHENTICATION_PARAM, this.getLogin(), this.getPassword());
+			httpPost.setEntity(new StringEntity(jsonPayload, ContentType.APPLICATION_JSON));
+
+			HttpResponse response = client.execute(httpPost);
+			int statusCode = response.getStatusLine().getStatusCode();
+
+			if (statusCode == 403) {
+				this.authenticationCookie = EntityUtils.toString(response.getEntity());
+				return false;
+			}
+			if (statusCode == 204) {
+				Header[] headers = response.getHeaders(HaivisionConstant.SET_COOKIE);
+				if (headers.length == 0) {
+					String responseBody = EntityUtils.toString(response.getEntity());
+					String sessionId = extractSessionId(responseBody);
+					if (sessionId != null) {
+						this.authenticationCookie = "id=" + sessionId;
+						System.out.println("Session ID retrieved: " + sessionId);
```

Comment:
This is an early implementation so most probably system.out calls will be changed to logger calls, but just leaving this comment here so it doesn't get overlooked